### PR TITLE
Feature/new button and storybook

### PR DESCRIPTION
**Pull request by henryfradley**
## Motivation

Add storybook for documentation and add `primary` button variant.

### Added

Added new token set and processed into `light-tokens.css`, `dark-tokens.css`, and `global-tokens.css`

### Screenshots/Videos

https://user-images.githubusercontent.com/68616911/184555931-45c120cd-f4c7-4c07-930b-e437950fe09a.mov

### CODE REVIEW

File: dist/index.d.ts [hunk 1]
```diff
@@ -1,14 +1,6 @@
...
 
-declare type vals = 'top' | 'left' | 'right' | 'bottom';
-interface CoordsTypes {
-    top: vals;
-    left: vals;
-    right: vals;
-    bottom: vals;
-}
-
 declare type integerVals$1 = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
```
Comment (jaredshaunsmith) [inline dist/index.d.ts]: @henryfradley What's the purpose of the `$1` here?

File: src/components/Theme/dark-tokens.module.css [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  --spectrum-yellow-400: #ffce4a;
+  --spectrum-yellow-500: #ffc424;
+  --spectrum-yellow-600: #ffad1f;
+  --spectrum-yellow-700: #fa8900;
+  --spectrum-yellow-800: #e55c00;
+  --spectrum-yellow-900: #c73f00;
+  --spectrum-yellow-1000: #752100;
+  --spectrum-yellow-000: #fff8e4;
+  --global-fontFamilies-ivypresto-display: IvyPresto Display;
+  --global-fontFamilies-sf-compact: SF Compact;
```
Comment (jaredshaunsmith) [inline src/components/Theme/dark-tokens.module.css]: do we need system UI defined here as well, or is the idea just to manually add that anywhere we need it?

File: dist/index.d.ts (hunk 1 above)
Comment (henryfradley) [reply dist/index.d.ts]: strange, I went through and refactored those types to be cleaner and this is gone.

File: src/components/Theme/dark-tokens.module.css (hunk 2 above)
Comment (henryfradley) [reply src/components/Theme/dark-tokens.module.css]: I modified the script to append `system-ui, sans-serif` to any fontFamily token!

File: src/utils/margins/margins.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
-const integerVals = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10
-const autoVals = 'auto'
+import type { integers } from '../lib/integers'
```
Comment (henryfradley) [inline src/utils/margins/margins.ts:2]: this type was really ending up as 'number' or 'string', now only 1-10 or auto will be allowed

**Comment by jaredshaunsmith** [comment]
We may not want that. We do have one serif font

**Comment by henryfradley** [comment]
> We may not want that. We do have one serif font
> […](#)
> Sent from my iPhone
> On Aug 19, 2022, at 7:52 PM, Henry Fradley ***@***.***> wrote: ﻿ @henryfradley commented on this pull request. In src/components/Theme/dark-tokens.module.css: > + --spectrum-green-900: #003d2e; + --spectrum-green-1000: #00291f; + --spectrum-green-000: #f3fcf9; + --spectrum-yellow-100: #ffeebe; + --spectrum-yellow-200: #ffe397; + --spectrum-yellow-300: #ffd971; + --spectrum-yellow-400: #ffce4a; + --spectrum-yellow-500: #ffc424; + --spectrum-yellow-600: #ffad1f; + --spectrum-yellow-700: #fa8900; + --spectrum-yellow-800: #e55c00; + --spectrum-yellow-900: #c73f00; + --spectrum-yellow-1000: #752100; + --spectrum-yellow-000: #fff8e4; + --global-fontFamilies-ivypresto-display: IvyPresto Display; + --global-fontFamilies-sf-compact: SF Compact; I modified the script to append system-ui, sans-serif to any fontFamily token! — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because your review was requested.

Okay, should we just append it to `SF Compact` and `Degular`?

File: src/components/Theme/light-tokens.module.css [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  --spectrum-yellow-400: #ffce4a;
+  --spectrum-yellow-500: #ffc424;
+  --spectrum-yellow-600: #ffad1f;
+  --spectrum-yellow-700: #fa8900;
+  --spectrum-yellow-800: #e55c00;
+  --spectrum-yellow-900: #c73f00;
+  --spectrum-yellow-1000: #752100;
+  --spectrum-yellow-000: #fff8e4;
+  --global-fontFamilies-ivypresto-display: IvyPresto Display;
+  --global-fontFamilies-sf-compact: 'SF Compact', system-ui, sans-serif;
```
Comment (henryfradley) [inline src/components/Theme/light-tokens.module.css:63]: Now I'm just adding the back up fonts to SF Compact, we can revisit this later and add backups for the other fonts if necessary. cc @jaredshaunsmith

**Review by jaredshaunsmith — APPROVED** [review]

**Pull request merged**